File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/registration-ecertificate.model.ts
export interface IRegistrationECertificateModel {
  id: string;
  userId: string;
  pdfFileName: string;
  base64Pdf: string;
  base64Image: string;
  width: number;
  height: number;
}

export class RegistrationECertificateModel implements IRegistrationECertificateModel {
  public id: string = '';
  public userId: string = '';
  public pdfFileName: string = '';
  public base64Pdf: string = '';
  public base64Image: string = '';
  public width: number = 0;
  public height: number = 0;

  constructor(data?: IRegistrationECertificateModel) {
    if (data != null) {
      this.id = data.id;
      this.userId = data.userId;
      this.pdfFileName = data.pdfFileName;
      this.base64Pdf = data.base64Pdf;
      this.base64Image = data.base64Image;
      this.width = data.width;
      this.height = data.height;
    }
  }
}

File: pdlearning/src/microservices/course-microservice/Api/Application/Models/RegistrationECertificateModel.cs
using System;
using Microservice.Course.Domain.Entities;

namespace Microservice.Course.Application.Models
{
    public class RegistrationECertificateModel
    {
        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public string PdfFileName { get; set; }

        public string Base64Pdf { get; set; }

        public string Base64Image { get; set; }

        public int Width { get; set; }

        public int Height { get; set; }

        public static RegistrationECertificateModel Create(RegistrationECertificate registrationECertificate)
        {
            return new RegistrationECertificateModel
            {
                Id = registrationECertificate.Id,
                UserId = registrationECertificate.UserId,
                PdfFileName = registrationECertificate.PdfFileName,
                Base64Image = registrationECertificate.Base64Image,
                Base64Pdf = registrationECertificate.Base64Pdf,
                Width = registrationECertificate.Width,
                Height = registrationECertificate.Height
            };
        }
    }
}
